commands/azureCommands/create-registry.ts

```diff
@@ -0,0 +1,210 @@
+
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { RegistryNameStatus } from "azure-arm-containerregistry/lib/models";
+import { ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { ResourceGroup } from "azure-arm-resource/lib/resource/models";
+import * as vscode from "vscode";
+import { reporter } from '../../telemetry/telemetry';
+import { AzureCredentialsManager } from '../../utils/azureCredentialsManager';
+const teleAzureId: string = 'vscode-docker.create.registry.azureContainerRegistry';
+const teleCmdId: string = 'vscode-docker.createRegistry';
+
+export async function createRegistry(): Promise<void> {
+    let subscription: SubscriptionModels.Subscription;
+    let resourceGroup: ResourceGroup;
+    let location: string;
+
+    try {
+        subscription = await acquireSubscription();
+        resourceGroup = await acquireResourceGroup(subscription);
+
+    } catch (error) {
+        return;
+    }
+    const client = AzureCredentialsManager.getInstance().getContainerRegistryManagementClient(subscription);
+
+    let registryName: string;
+    try {
+        registryName = await acquireRegistryName(client);
+    } catch (error) {
+        return;
+    }
+
+    const sku: string = await acquireSKU();
+    location = await acquireLocation(resourceGroup, subscription);
+
+    client.registries.beginCreate(resourceGroup.name, registryName, { 'sku': { 'name': sku }, 'location': location }).then((response): void => {
+        vscode.window.showInformationMessage(response.name + ' has been created succesfully!');
+    }, (error): void => {
+        vscode.window.showErrorMessage(error.message);
+    })
+
+    //Acquiring telemetry data here
+    if (reporter) {
+        /* __GDPR__
+           "command" : {
+              "command" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+           }
+         */
+        reporter.sendTelemetryEvent('command', {
+            command: teleCmdId
+        });
+
+        if (registryName.toLowerCase().indexOf('azurecr.io')) {
+            /* __GDPR__
+               "command" : {
+                  "command" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+               }
+             */
+            reporter.sendTelemetryEvent('command', {
+                command: teleAzureId
+            });
+        }
+    }
+
+}
+async function acquireSKU(): Promise<string> {
+    let skus: string[] = ["Basic", "Standard", "Premium"];
+    let sku: string;
+    sku = await vscode.window.showQuickPick(skus, { 'canPickMany': false, 'placeHolder': 'Choose a SKU' });
+    if (sku === undefined) { throw new Error('User exit'); }
```

Comment:
> { throw new Error('User exit'); } [](start = 26, length = 34)

Not sure if this style is readable. Is there a requirement to run tslint or something similar?